docs/internal/types.md

```diff
@@ -0,0 +1,236 @@
+
+Fields can be initialised in two ways:
+ * `var`: Must be initialised on construction, but can change after.
+ * `let`: Must be initialised on construction, and cannot change later.
+
+They can also be a private member, if its name starts with `_`.
+
+They can also have a keyword `embed`, which means ...
+
+### Methods
+
+Methods are functions that belong to a class or interface. Like functions, they can have arguments and a return type.
+
+Methods can be called statically (via `Class::method()`) or dynamically (via `object.method()`).
+On the static case, the `self` pointer is a descriptor forthe type, while on the dynamic case, it's a descriptor to the instance (and its vtable, type info, etc).
+
+Static calls cannot change the fields of the class unless an object is being created and returned, like in the case of constructors.
+
+### Constructors and Destructors
+
+Verona distinguishes memory allocation with object construction.
+There are no `constructors` per se, in Verona, only static methods that can allocate memory (using `new`), build an object and return it.
+
+Example:
+```ts
+class Capibara
+{
+  let name : String;
+  var weight : F32; // Kgs
+
+  // Create a baby capibara
+  newborn(name : String) : Capibara
+  {
+    var self : Capibara = new Capibara;
+    self.name = name;
+    self.weight = 3; // Tiny baby!
+    self; // Returns `self`
+  }
+
+  eat(self: Capibara, food : F32)
+  {
+    self.weight += food;
+  }
+
+  poop(self: Capibara)
+  {
+    self.weight -= 0.1; // Tiny poop!
+  }
+};
+```
+
+On the example above, the `newborn` method creates a `Capibara` with a specific name and a default weight.
+The `name` can't change, so we declare it with `let`, while the `weight` does change, so it's a `var`.
+
+There's nothing special about the method `newbord` in comparison with others, apart from the fact that it allocates, initialises and returns an object of the same type.
+
+Destructors are still under debate, but they won't have to deallocate memory like in C++, just execute finalization tasks like closing file handles, running some cleanup routines, checking final consistency, etc.
+
+## Primitive Types
+
+Verona treats numeric types specially in the compiler but their representaiton isn't special in the language.
+
+Verona's numeric types are singleton types (no fields) that have their machine representaiton based on their names.
+So, `I32` is a signed integer with 32-bit width (in LLVM's parlour, an `i32`) and so on.
+
+The full list of primitive numeric types are:
+ * Unsigned integers: `U8`, `U16`, `U32`, `U64` and `U128`
+ * Signed integers: `I8`, `I16`, `I32`, `I64` and `I128`
+ * Floating Point: `F32` and `F64`
+ * System types: `ULength` and `ILength` (machine-word sized integers)
+
+Note, integer literals have no defined type and their types (and representation) will be inferred by the compiler depending on the type of the variable being assigned to.
```

Comment:
Im not sure what this means. An example would be good.

`let x = 1`

what's the inferred type?

Reply:
Good point. `let x = 1` is syntax error, but `let x : I32 = 1` and `let x : U64 = 1` infer different types for `1`.